.github/workflows/ci.yml

```diff
@@ -20,39 +20,19 @@ jobs:
         python-version: ["3.12"]
 
     steps:
-      - uses: "actions/checkout@v4"
+      - name: "Checkout source files"
+        uses: "actions/checkout@v4"
       - name: Setup environment
         uses: ./.github/actions/setup
         with:
           python-version: ${{ matrix.python-version }}
           cache-pre-commit: true
           uv-version: ${{ env.UV_VERSION }}
           uv-install-options: "--all-extras"
-      - name: "Check supported device md files are up to date"
-        run: |
-          uv run pre-commit run generate-supported --all-files
-      - name: "Linting and code formating (ruff)"
-        run: |
-          uv run pre-commit run ruff --all-files
-      - name: "Typing checks (mypy)"
-        run: |
-          source .venv/bin/activate
-          pre-commit run mypy --all-files
-      - name: "Run trailing-whitespace"
-        run: |
-          uv run pre-commit run trailing-whitespace --all-files
-      - name: "Run end-of-file-fixer"
-        run: |
-          uv run pre-commit run end-of-file-fixer --all-files
-      - name: "Run check-docstring-first"
-        run: |
-          uv run pre-commit run check-docstring-first --all-files
-      - name: "Run debug-statements"
-        run: |
-          uv run pre-commit run debug-statements --all-files
-      - name: "Run check-ast"
+
+      - name: "Run pre-commit checks"
```

Comment:
Looks like we were missing the `ruff format`, `check-yaml` and `doc8` from the CI. It's probably better to just run all the checks in a single step and pass `--verbose` to avoid having to keep them in sync.

Reply:
Yes, no need to repeat oneself here, I have been wanting to do this for a while :+1: